pywbemtools/pywbemcli/_cmd_connection.py

```diff
@@ -125,27 +133,40 @@ def connection_select(context, name):
     connection. If the NAME argument is omitted, prompt for selecting one of
     the connection definitions in the connections file.
 
-    This command is useful in the interactive mode of pywbemcli.
+    Selects a connection from the persistently stored set of named connections
+    if NAME exists in the store. If NAME not supplied, a list of connections
+    from the connections definition file is presented with a prompt for the
+    user to select a connection.
```

Comment:
The previous paragraph (starting on line 132) already describes this. The new paragraph here has no new information and can be removed.

Reply:
What was missing is that this really does two things.  Sets current and also sets default in the connections file.  That is where some of the complexity comes in.

First paragraph now talks about the NAME and what happes if you do not provide it.

Paragraph 2 about setting it as default and also as current.

I actually do not like this combination of 2 things (one persistent and one transient) done at the same time. I would rather that we could separate so it either set NAME to be the default in the store or set NAME to be the current.  SUGGESTIONS WELCOME.  How about an option that would set the persistent default.

**DISCUSSION**  Or just tell me if this makes sense.
connection select abc     only sets it to current
connection select abc -d  sets the default-connection in the connections file.

Reply:
I went ahead and added an option --default that sets both default in the connections file and current. If --default not set, the name is only set to current and the persistent default not changed.

Reply:
See discussion point.

Reply:
**DISCUSSION**
That depends on whether you think the users will try to use the names as some sort of meaning.  Thus, I always set the simplest mock as the name mock1. The minute I use the concept of name changing to set the default, that sense is lost.  

I agree that there is an issue in that you cannot remove the default  connection definition completely without removing the connection in the current code.  However that is easily changed. Further, the use of the default is clearly a specific behavior. The user must start pywbemcli with NO connection definition at all before the default is enabled.  NOTE: This is a good place for a mesage in verbose mode.  "Connect xxxx (default) is current".

As far as hard to distinguish, that is easily resolvable; different symbol, col in table for these states, etc.

I really think this scheme works but have no issue with a clearer way to express what is the default in the table output, etc.

Reply:
I can buy your argument that connection definition names should have a meaning. However, I also believe mthat having a concept of a default connection complicates things more than needed.

What the (any form of) persisted default provides is the ability to omit `--name emc3` on the command line. When in interactive mode, you already have the current connection that serves as the default so there it is not even an advantage. In fact, it complicates things because one needs to understand whether the current connection or the persisted default connection has precedence.

My suggestion would be to drop the concept of a persisted default altogether.

Reply:
We agreed it is worth it. The precedence difficulty is handled by pywbemcli outputting the name of the connection that is used if it is invoked without one.